### PR TITLE
com.github.zyedidia.micro

### PR DESCRIPTION
**Pull request by blt-r**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
 `--talk-name=org.freedesktop.Flatpak` is allowed for IDEs and micro has integrated terminal 
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/zyedidia/micro/issues/2546#issuecomment-1499778098**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
Fixed appdata file is already merged into upstream, next release it will be gone

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.zyedidia.micro.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+app-id: com.github.zyedidia.micro
```
Comment (hfiguiere) [inline com.github.zyedidia.micro.yml:1]: app-id must start with `io.github` as per the docs.

File: com.github.zyedidia.micro.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        url: 'https://github.com/zyedidia/micro.git'
+        tag: 'v2.0.11'
+        commit: '225927b9a25f0d50ea63ea18bc7bb68e404c0cfd'
+    
+      - type: file
+        path: modules.txt
+        dest: vendor
+      - 'go.mod.yml'
+
+      # Original metainfo file is broken, see https://github.com/zyedidia/micro/pull/2792
```
Comment (hfiguiere) [inline com.github.zyedidia.micro.yml:88]: a patch is always better.

File: com.github.zyedidia.micro.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	<developer_name>Zachary Yedidia</developer_name>
+	<screenshots>
+		<screenshot type="default">
+			<caption>Micro Text Editor editing its source code.</caption>
+			<image type="source">https://raw.githubusercontent.com/zyedidia/micro/master/assets/micro-solarized.png</image>
+		</screenshot>
+	</screenshots>
+	<url type="homepage">https://micro-editor.github.io</url>
+	<url type="bugtracker">https://github.com/zyedidia/micro/issues</url>
+</component>
```
Comment (hfiguiere) [inline com.github.zyedidia.micro.metainfo.xml:24]: missing `content_rating`

File: com.github.zyedidia.micro.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component>
+	<id>com.github.zyedidia.micro</id>
+	<launchable type="desktop-id">micro.desktop</launchable>
+	<name>Micro Text Editor</name>
+	<summary>A modern and intuitive terminal-based text editor</summary>
+	<metadata_license>MIT</metadata_license>
```
Comment (hfiguiere) [inline com.github.zyedidia.micro.metainfo.xml:7]: no project license?

File: com.github.zyedidia.micro.metainfo.xml (hunk 3 above)
Comment (blt-r) [reply com.github.zyedidia.micro.metainfo.xml]: Added it

File: com.github.zyedidia.micro.yml (hunk 2 above)
Comment (blt-r) [reply com.github.zyedidia.micro.yml]: Now it is a patch

File: com.github.zyedidia.micro.yml (hunk 1 above)
Comment (blt-r) [reply com.github.zyedidia.micro.yml]: Now it is `io.github.zyedidia.micro`

File: com.github.zyedidia.micro.metainfo.xml (hunk 4 above)
Comment (blt-r) [reply com.github.zyedidia.micro.metainfo.xml]: Added it

**Comment by blt-r** [comment]
Hopefully my PR will get merged into upstream and then everything will be good

**Comment by blt-r** [comment]
Ok it's merged, now everything is good

**Comment by hfiguiere** [comment]
bot, build io.github.zyedidia.micro

**Comment by blt-r** [comment]
will need to merge https://github.com/flathub/flatpak-builder-lint/pull/132, so it has flatpak-spawn access (which allowed for IDEs/code-editors)

**Comment by blt-r** [comment]
Should be fixed now

**Comment by blt-r** [comment]
Looks to be working to me 👍

**Pull request closed without merging**